test/shape.cpp

```diff
@@ -449,12 +449,17 @@ TEST(shape_conversion) {
 
 TEST(shape_transpose) {
   dense_shape<3> s(3, 5, 8);
-  shape<dim<>, dim<>, dense_dim<>> transposed = transpose<1, 2, 0>(s);
-  ASSERT_EQ(transposed.template dim<0>().extent(), 5);
-  ASSERT_EQ(transposed.template dim<1>().extent(), 8);
-  ASSERT_EQ(transposed.template dim<2>().extent(), 3);
+  shape<dim<>, dim<>, dense_dim<>> transposed_120 = transpose<1, 2, 0>(s);
+  ASSERT_EQ(transposed_120.template dim<0>().extent(), 5);
```

Comment:
Do you need `template` here? I think just `.dim<0>()` works.

Reply:
Oh, you're right!! I think maybe because transposed_120 is not a template type parameter, it works. I will update the docs too, I think it would be more clear to use `dim<n>` than x, y, z, etc.